### ISSUE TITLE
MT-M82 unknown  Khmer character text

### ISSUE THREAD

**Issue by phearunum** [mike42/escpos-php#856]
![70766351_981133362278750_8790267075164635136_n](https://user-images.githubusercontent.com/35354771/71305342-eba2b700-2404-11ea-92cd-4e6c0a15380d.jpg)
![73247005_404924387114772_5082432734924636160_n](https://user-images.githubusercontent.com/35354771/71305343-ec3b4d80-2404-11ea-87a2-83bfc352b412.jpg)

**Comment by mike42** [comment]
I don't think that the Epson TM-T82II is capable of printing Khmer by itself. The documentation I checked was:

- The documentation for `ESC t`: [here](https://reference.epson-biz.com/modules/ref_escpos/index.php?content_id=32) (I don't believe any of those code pages contain Khmer characters, correct me if I'm wrong)
- The code page support documentation for each model of Epson printer: [here](https://reference.epson-biz.com/modules/ref_charcode_en/index.php?content_id=117)

This library is correctly substituting "?" for characters which it cannot determine how to send.

I've recently been updating the [Text encoding](https://github.com/mike42/escpos-php/wiki/Text-encoding) documentation on our Wiki to include work-arounds for this type of situation.

Two which may be worth a look are:

- Find and use an external tool to render the receipt, such as a HTML-to-image converter (we can print images!). There is an example on the wiki which shows how `wkhtmltoimage` can help.
- Use `ImagePrintBuffer` to write each line of text as an image using Imagick. I don't currently have time to see whether this will work for Khmer script with this layout, but it might be worth attempting a "Hello World" receipt, and building from there if it works. Again, there is an example on the wiki.

**Comment by phearunum** [comment]
Hi Mike42,

I have try to use ImagePrintBuffer  but still print with wrong character word 

try {
             $connector = new \Escpos\PrintConnectors\NetworkPrintConnector('192.168.123.100','9100');
             $profile = \Escpos\CapabilityProfile::load("default");
             $buffer = new \Escpos\PrintBuffers\ImagePrintBuffer();
             $buffer -> setFont(APPPATH .'third_party/Mike42/Escpos/resources/KhmerOS.ttf');
             $buffer->setFontSize('20');
             $printer = new \Escpos\Printer($connector, $profile);
                 $printer -> setPrintBuffer($buffer);
                 $printer -> text("ប៉ិបស៊ី ");
                $printer -> text("កូកាកូឡា");
                 $printer -> text("អូអីឈីក");
                $printer -> text("ប៉ូការី ស្វេតដប ");
             $printer -> cut();
             $printer->feed(2);
         } catch (Exception $e) {
             echo "Couldn't print to this printer: " . $e -> getMessage() . "\n";
         } finally {
             $printer -> close();
         }

![image](https://user-images.githubusercontent.com/35354771/83944674-8bc38f80-a82f-11ea-8810-6cc6e9546d4b.png)

Could please help me to resole or any recommend  about Khmer character encoding with print ImagePrintBuffer  ?
Big Thank ,
